Add note concerning Podman/MacOS combination; (#435)
paritytech/zombienet: README.md

Internally zombienet is a `javascript` library, designed to run on `Node.js` and support different
backend `providers` to run the *nodes*, at this moment `kubernetes`, `podman` and `native` are
supported. 

> **Note:** `Podman` is supported in MacOS as an app but this is happening through an
> internal vm. The provider's code in zombienet supports `podman` running native - thus `podman` for
> `zombienet` can be used only on linux. This is related, mostly, to paths and directories used by
> store configuration (chain-spec) and the data directory.

Reviewer: ```suggestion
**Note:** Currently, it is only possible to use `podman` for Zombienet users on Linux machines. 
Although `podman` comes with support for macOS, it is done using an internal VM and the Zombienet provider code expects `podman` to be running natively.
```

*without* having `Node.js` installed **but** each `provider` defines it's own requirements (e.g.
`k8s`, `podman`).

> **Note:** `Podman` is supported in MacOS as an app but this is happening through an
> internal vm. The provider's code in zombienet supports `podman` running native - thus `podman` for
> `zombienet` can be used only on linux. This is related, mostly, to paths and directories used by
> store configuration (chain-spec) and the data directory.

Reviewer: ```suggestion
**Note:** Currently, it is only possible to use `podman` for Zombienet users on Linux machines. 
Although `podman` comes with support for macOS, it is done using an internal VM and the Zombienet provider code expects `podman` to be running natively.
```

Zombienet support [Podman](https://podman.io/) *rootless* as provider, you only need to have
`podman` installed in your environment to use and either set in the *network* file or with the

> `--provider` flag in the cli. `Podman` for `zombienet` can be used only on linux even though `podman`
> is an app supported also on MacOS. This is related, mostly, to paths and directories used by 
> store configuration (chain-spec) and the data directory.

Reviewer: ```suggestion
`--provider` flag in the cli. `Podman` for `zombienet` is currently only supported for Linux machines.
This is mostly related to paths and directories used by 
store configuration (chain-spec) and the data directory.
```
